src/api/accounts.rs

```diff
@@ -0,0 +1,515 @@
+    where
+        Self: Sized,
+    {
+        Accounts { transport }
+    }
+
+    fn transport(&self) -> &T {
+        &self.transport
+    }
+}
+
+impl<T: Transport> Accounts<T> {
+    /// Gets the parent `web3` namespace
+    fn web3(&self) -> Web3<T> {
+        Web3::new(self.transport.clone())
+    }
+
+    /// Signs an Ethereum transaction with a given private key.
+    pub fn sign_transaction(&self, tx: TransactionParameters, key: &SecretKey) -> SignTransactionFuture<T> {
+        SignTransactionFuture::new(self, tx, key)
+    }
+
+    /// Hash a message. The data will be UTF-8 HEX decoded and enveloped as
+    /// follows: `"\u{19}Ethereum Signed Message:\n" + message.length + message`
+    /// and hashed using keccak256.
+    pub fn hash_message<S>(&self, message: S) -> H256
+    where
+        S: AsRef<str>,
+    {
+        let message = message.as_ref();
+        let eth_message = format!("\u{0019}Ethereum Signed Message:\n{}{}", message.len(), message);
+
+        eth_message.as_bytes().keccak256().into()
+    }
+
+    /// Sign arbitrary string data. The data is UTF-8 encoded and enveloped the
+    /// same way as with `hash_message`.
+    pub fn sign<S>(&self, message: S, key: &SecretKey) -> Result<SignedData, Error>
+    where
+        S: AsRef<str>,
+    {
+        let message = message.as_ref().to_string();
+        let message_hash = self.hash_message(&message);
+
+        let signature = key.sign(&message_hash[..]).map_err(EthsignError::from)?;
+        // this is what web3.js does ¯\_(ツ)_/¯, it is documented here:
+        // https://web3js.readthedocs.io/en/v1.2.2/web3-eth-accounts.html#sign
+        let v = signature.v + 27;
+
+        let signature_bytes = Bytes({
+            let mut bytes = Vec::with_capacity(65);
+            bytes.extend_from_slice(&signature.r[..]);
+            bytes.extend_from_slice(&signature.s[..]);
+            bytes.push(v);
+            bytes
+        });
+
+        Ok(SignedData {
+            message,
+            message_hash: message_hash.into(),
+            v,
+            r: signature.r.into(),
+            s: signature.s.into(),
+            signature: signature_bytes,
+        })
+    }
+
+    /// Recovers the Ethereum address which was used to sign the given data.
+    pub fn recover<S, Sig>(&self, message: S, signature: Sig) -> Result<Address, Error>
+    where
+        S: AsRef<str>,
+        Sig: IntoSignature,
+    {
+        let message_hash = self.hash_message(message);
+        let signature = signature.into_signature();
+
+        let public_key = signature.recover(&message_hash[..]).map_err(EthsignError::from)?;
+
+        Ok(public_key.address().into())
+    }
+}
+
+type MaybeReady<T, R> = Either<FutureResult<R, Error>, CallFuture<R, <T as Transport>::Out>>;
+
+type TxParams<T> = Join3<MaybeReady<T, U256>, MaybeReady<T, U256>, MaybeReady<T, String>>;
+
+/// Future resolving when transaction signing is complete
+pub struct SignTransactionFuture<T: Transport> {
```

Comment:
I can change this to `impl Trait` if you feel strongly about it, but I don't think `async`/`await` is a good idea here because that would mean the crate would mix futures 0.1 and std futures depending on what is being called.

Reply:
Not feeling strong, the codebase is not migrated indeed. I'm fine with the futures as it is currently.